express-post-json/index.js

```diff
@@ -0,0 +1,28 @@
+const express = require('express');
+const app = express();
+let nextId = 1;
+
+const grades = {
+
+};
+
+app.get('/api/grades', (req, res) => {
+  const array = [];
+  array.push(grades[nextId]);
+  res.json(array);
+});
+
+const jsonMiddleware = express.json();
+app.use(jsonMiddleware);
+
+app.post('/api/grades', (req, res) => {
+  req.body.id = nextId;
+  nextId++;
+  grades[nextId] = req.body;
```

Comment:
This will assign the wrong property to the new grade, you want to use it, then increment it:
```JAVASCRIPT
 grades[nextId] = req.body;
nextId++;
```